tests/source/lhs-to-rhs-between-comments/assignment.rs

```diff
@@ -0,0 +1,48 @@
+// "=" Assignemnt
+fn main () {
+    let var = first;
+    var = second;
+    }
+    fn main () {
+    let var = /* Block comment */ first;
+    var = /* Block comment */ second;
+    }
+    fn main () {
+    let var = // Line comment
+    first;
+        var = // Line comment
+        second;
+    }
+    fn main () {
+    let var = /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ first;
+    var = /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ second;
+    }
+    fn main () {
+    let var = /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ first;
+    var = /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ second;
+    }
+    
+    // BinOp Assigment
+    fn main () {
+    let var = first;
+    var += second;
+    }
+    fn main () {
+    let var = /* Block comment */ first;
+    var -= /* Block comment */ second;
+    }
+    fn main () {
+    let var = // Line comment
+    first;
+    var *= // Line comment
+    second;
+    }
+    fn main () {
+    let var = /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ first;
+    var /= /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ second;
+    }
+    fn main () {
+    let var = /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ first;
+    var /= /* Block comment longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg */ second;
+    }
+    
```

Comment:
Solid test suite! For the sake of extreme thoroughness, do you think it'd be worthwhile to add a multiline blockstyle comment case?

Reply:
Added multiline comments test cases.

However, I think there is an **issue** with the multiline comments formatting by `combine_strs_with_missing_comments()` (not caused by this PR so I didn't try to fix it here).  An example of the issue is that the following formatted code:
```rust
let var = /* Block comment line 1
    * Block comment line 2 */
    first;
```
Should probably have been **either**:
```rust
let var = /* Block comment line 1
           * Block comment line 2 */
    first;
```
**or**:
 ```rust
let var = /* Block comment line 1
            * Block comment line 2 */  first;
```
**Is this really an issue?  If yes, which of the output versions is desired?** If this is an issue that should be fixed, I can try to do that.

Reply:
> Is this really an issue?

it is _an_ issue yes. some of these comment edge cases can be so exhausting sometimes :smile: it would have to be the former or what we did in other similar scenarios 

e.g. something like this (hand typed to give a visual, indentation may be off)
```rust
let var = 
    /* Block comment line 1
     * Block comment line 2 */
    first;
```

I'd prefer the former in the spirit of maintaining developer defined comment association, as odd as such a comment would be in these types of local cases.